pkg/kubelet/dockertools/manager.go

```diff
@@ -1336,7 +1336,7 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 			glog.Errorf("Failed tearing down the infra container: %v", err)
 			errs <- err
 		}
-		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod); err != nil {
+		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod, "Need to kill pod."); err != nil {
```

Comment:
This pr focus on container killing reason. It describes details if container will be restart or re-created after being killed. "will not be restart" may be omitted.